Marlin/src/lcd/e3v2/proui/dwin.cpp

```diff
@@ -162,7 +162,27 @@
 #define DWIN_SCROLL_UPDATE_INTERVAL      SEC_TO_MS(2)
 #define DWIN_REMAIN_TIME_UPDATE_INTERVAL SEC_TO_MS(20)
 
-#define BABY_Z_VAR TERN(HAS_BED_PROBE, probe.offset.z, HMI_data.ManualZOffset)
+#if HAS_MESH
+  #define BABY_Z_VAR TERN(HAS_BED_PROBE, probe.offset.z, HMI_data.ManualZOffset)
+
+  // These may be overridden in Configuration.h if a smaller area is desired
+  // AUTO_BED_LEVELING_BILINEAR does not define MESH_INSET
+  #ifndef MESH_MIN_X
+    #define MESH_MIN_X (_MAX(X_MIN_BED + PROBING_MARGIN, X_MIN_POS))
+  #endif
+  #ifndef MESH_MIN_Y
+    #define MESH_MIN_Y (_MAX(Y_MIN_BED + PROBING_MARGIN, Y_MIN_POS))
+  #endif
+  #ifndef MESH_MAX_X
+    #define MESH_MAX_X (_MIN(X_MAX_BED - (PROBING_MARGIN), X_MAX_POS))
+  #endif
+  #ifndef MESH_MAX_Y
+    #define MESH_MAX_Y (_MIN(Y_MAX_BED - (PROBING_MARGIN), Y_MAX_POS))
+  #endif
```

Comment:
In ABL Bilinear these are variables, not constants, so it is dangerous to assume any constant value.